internal/provider/access_policy_resource.go

```diff
@@ -261,90 +316,144 @@ func (r *accessPolicyResource) ImportState(ctx context.Context, req resource.Imp
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
 
-func convertAccessPolicyModelToPolicyDTO(model accessPolicyResourceModel, externalID *string) aembit.PolicyDTO {
-	var policy aembit.PolicyDTO
+func convertAccessPolicyModelToPolicyDTO(model accessPolicyResourceModel, externalID *string) aembit.CreatePolicyDTO {
+	var policy aembit.CreatePolicyDTO
 	policy.EntityDTO = aembit.EntityDTO{
 		Name:     model.Name.ValueString(),
 		IsActive: model.IsActive.ValueBool(),
 	}
 	policy.ClientWorkload = model.ClientWorkload.ValueString()
 	policy.ServerWorkload = model.ServerWorkload.ValueString()
-	policy.CredentialProvider = model.CredentialProvider.ValueString()
+
+	policy.CredentialProviders = make([]aembit.PolicyCredentialMappingDTO, len(model.CredentialProviders))
+
+	if len(model.CredentialProvider.ValueString()) > 0 {
+		policy.CredentialProviders = make([]aembit.PolicyCredentialMappingDTO, 1)
+		policy.CredentialProviders[0] = aembit.PolicyCredentialMappingDTO{
+			PolicyId:             "00000000-0000-0000-0000-000000000000",
+			CredentialProviderId: model.CredentialProvider.ValueString(),
+			MappingType:          "None",
+			AccountName:          "",
+			HeaderName:           "",
+			HeaderValue:          "",
+			HttpbodyFieldPath:    "",
+			HttpbodyFieldValue:   "",
+		}
+	} else {
+		policy.CredentialProviders = make([]aembit.PolicyCredentialMappingDTO, len(model.CredentialProviders))
+
+		for i, credentialProvider := range model.CredentialProviders {
+			policy.CredentialProviders[i] = aembit.PolicyCredentialMappingDTO{
+				PolicyId:             "00000000-0000-0000-0000-000000000000",
```

Comment:
Is PolicyId always this zero value? Is it calculated or generated elsewhere?

Reply:
In create or update API does not use the PolicyId but we cant pass null or empty too. It should be default.
I tried not to send it at all but server return 400.

So I am passing the default GUID value.